idx/query_matcher_test.go

```diff
@@ -0,0 +1,140 @@
+// Copyright (c) 2018 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package idx_test
+
+import (
+	"testing"
+
+	"github.com/m3db/m3ninx/idx"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestQueryMatcherTermQuery(t *testing.T) {
+	for _, tc := range []struct {
+		left     idx.Query
+		right    idx.Query
+		expected bool
+	}{
+		{
+			left:     idx.NewTermQuery([]byte("abc"), []byte("def")),
+			right:    idx.NewTermQuery([]byte("abc"), []byte("def")),
+			expected: true,
+		},
+		{
+			left:     idx.NewTermQuery([]byte("abc"), []byte("def")),
+			right:    idx.NewTermQuery([]byte("abc1"), []byte("def")),
+			expected: false,
+		},
+		{
+			left:     idx.NewTermQuery([]byte("abc"), []byte("def")),
+			right:    idx.NewTermQuery([]byte("abc"), []byte("def1")),
+			expected: false,
+		},
+	} {
+		require.Equal(t, tc.expected, idx.NewQueryMatcher(tc.left).Matches(tc.right))
+		require.Equal(t, tc.expected, idx.NewQueryMatcher(tc.right).Matches(tc.left))
+	}
+}
+
+func TestQueryMatcherTermMismatch(t *testing.T) {
```

Comment:
nit: isn't this covered by the test above?

Reply:
Sure, will remove